### PR TITLE
Move unify vars logic into binding manager.

### PR DESCRIPTION
**Pull request by dhatch**
PR checklist:

- [ ] Added changelog entry.

### CODE REVIEW

File: polar-core/src/bindings.rs [hunk 1]
```diff
@@ -43,13 +47,135 @@ pub struct BindingManager {
...
+            self.bind_variables(var, symbol);
+        } else {
+            self.add_binding(var, val);
+        }
+    }
+
+    /// Bind two variables together.
+    fn bind_variables(&mut self, left: &Symbol, right: &Symbol) {
+        match (self.variable_state(left), self.variable_state(right)) {
+            (VariableState::Bound(_), VariableState::Unbound) => {
```
Comment (samscott89) [inline polar-core/src/bindings.rs:92]: Oh, these are generally being handled in `unify_vars`, right?

File: polar-core/src/bindings.rs [hunk 2]
```diff
@@ -43,13 +47,135 @@ pub struct BindingManager {
...
+            }
+            (VariableState::Partial(_), VariableState::Bound(right_value)) => {
+                self.add_constraint(&op!(Unify, term!(left.clone()), right_value).into_term())
+                    .unwrap();
+            }
+            (VariableState::Partial(_), _) | (_, VariableState::Partial(_)) => {
+                self.add_constraint(
+                    &op!(Unify, term!(left.clone()), term!(right.clone())).into_term(),
+                )
+                .unwrap();
```
Comment (samscott89) [inline polar-core/src/bindings.rs:173]: Lots of unwraps here.

Comment (samscott89) [reply polar-core/src/bindings.rs:173]: Oh I see the TODO now

**Review by samscott89 — COMMENTED** [review]
This is hard to follow. I'm struggling to review the behaviour because I'm not sure what it's supposed to be doing.

File: polar-core/src/bindings.rs [hunk 3]
```diff
@@ -43,13 +47,135 @@ pub struct BindingManager {
...
+    }
+
+    /// Bind two variables together.
+    fn bind_variables(&mut self, left: &Symbol, right: &Symbol) {
+        match (self.variable_state(left), self.variable_state(right)) {
+            (VariableState::Bound(_), VariableState::Unbound) => {
+                // Replace binding.
+                self.add_binding(left, term!(right.clone()));
+            }
+            (VariableState::Unbound, VariableState::Bound(_)) => {
```
[the remark this answers is not in the document]
Comment (dhatch) [reply polar-core/src/bindings.rs:96]: In this PR, I handed this in the caller, but it's been changed on the latest: https://github.com/osohq/oso/pull/723/files#diff-cd1a8a2306ff7f13d326cbddac6d5d51a125e091b837906fbae6af2e18f0a99bR387

File: polar-core/src/bindings.rs (hunk 1 above)
Comment (dhatch) [reply polar-core/src/bindings.rs:92]: Yes, all used to be in unify vars. Alex removed here https://github.com/osohq/oso/pull/723/commits/ba82907976a42e9baf15a6e1a37d905d86c222b3
